pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -401,21 +401,42 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun, get
 		message := fmt.Sprintf("PipelineRun %s/%s awaiting remote resource", pr.Namespace, pr.Name)
 		pr.Status.MarkRunning(ReasonResolvingPipelineRef, message)
 		return nil
-	case errors.Is(err, trustedresources.ErrResourceVerificationFailed):
-		message := fmt.Sprintf("PipelineRun %s/%s referred pipeline failed signature verification", pr.Namespace, pr.Name)
-		pr.Status.MarkFailed(ReasonResourceVerificationFailed, message)
-		return controller.NewPermanentError(err)
 	case err != nil:
 		logger.Errorf("Failed to determine Pipeline spec to use for pipelinerun %s: %v", pr.Name, err)
 		pr.Status.MarkFailed(ReasonCouldntGetPipeline,
 			"Error retrieving pipeline for pipelinerun %s/%s: %s",
 			pr.Namespace, pr.Name, err)
 		return controller.NewPermanentError(err)
-	default:
-		// Store the fetched PipelineSpec on the PipelineRun for auditing
-		if err := storePipelineSpecAndMergeMeta(ctx, pr, pipelineSpec, pipelineMeta); err != nil {
-			logger.Errorf("Failed to store PipelineSpec on PipelineRun.Status for pipelinerun %s: %v", pr.Name, err)
+	}
+
+	// Skip the verification if it is not inline pipeline;
+	if pr.Spec.PipelineSpec == nil && pr.Status.PipelineSpec == nil {
+		var pipelineToBeVerified *v1beta1.Pipeline
+		if pipelineMeta != nil {
+			pipelineToBeVerified = &v1beta1.Pipeline{
+				ObjectMeta: *pipelineMeta.ObjectMeta,
+				Spec:       *pipelineSpec,
+			}
 		}
+		vp, err := c.verificationPolicyLister.VerificationPolicies(pr.Namespace).List(labels.Everything())
+		if err != nil {
+			return fmt.Errorf("failed to list VerificationPolicies from namespace %s with error %w", pr.Namespace, err)
+		}
+		var refSourceURI string
+		if pipelineMeta.RefSource != nil {
+			refSourceURI = pipelineMeta.RefSource.URI
+		}
```

Comment:
This looks duplicated with the other verify calls. Should we just move this into the verify func if it's always needed?